### ISSUE TITLE
Migrate content-server to react

### ISSUE THREAD

**Issue by dannycoates** [mozilla/fxa#3430]
it's the future.

todo: convert to epic once we have more clarity

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-561)
┆Issue Number: FXA-561

**Comment by lmorchard** [comment]
FWIW, this might be worth writing up as an ADR!

Also might be handy putting some thought into why React and not something else [like web components](https://briangrinstead.com/blog/firefox-webcomponents/), Svelte, Vue, etc. Obviously, I'm a fan of React and picked it for payments-server, but would be good to show our work in an ADR. (Or, maybe somehow even be surprised that React isn't the right choice?)

**Comment by shane-tomlinson** [comment]
Oooh, when I think about a total conversion away from Backbone, I am filled with fear and dread. Putting elements on the screen isn't the actual problem with a full conversion, it's all the background infrastructure to support:

* A/B tests
* metrics
* integrations with browsers
* input validation
* query param validation
* screen->screen state machine depending on the integration being used
* account management

There was a *lot* of blood, sweat, and tears shed to get where we are.

I think the only reasonable way to do this sort of work is to split it into chunks. settings first since it doesn't have to deal with different integrations as much and screen to screen transitions are consistent across integrations. Or maybe the password reset flow since its so simple. Update the views while keeping backbone models to do some things like manage integrations and accounts.

Granted, the flow *is* getting much simpler with the replacement of links with codes and email-first only. Once all that work is complete and the tests are updated, AND if we actively deprecate support for old browsers (perhaps based on the browser version returned in the fxaccounts:fxa_status WebChannel message) the screen->screen state machine should be *significantly* simpler and a large scale conversion of the signin/signup flows will be significantly easier.

**Comment by lmorchard** [comment]
> There was a lot of blood, sweat, and tears shed to get where we are.
> I think the only reasonable way to do this sort of work is to split it into chunks. 

Yeah, I think this is the right way to think about it. We should only re-implement with React if & where & when it simplifies things - and *definitely* with care to preserve the wisdom encoded in the system so far.

**Comment by data-sync-user** [comment]
➤ Jared Hirsch commented:

We're still talking about this, but not in this bug. Closing
